### ISSUE TITLE
Original files vs. truncated re-encoded files

### ISSUE THREAD

**Issue by hbredin** [SEERNET/pyannote-db-callhome#2]
Thanks for this plugin. That's great!

I'd rather use the original files rather that their truncated re-encoded versions.
This would make the plugin easier for anyone to use:

1. Download original files from talkbank.org
2. Edit ~/.pyannote/db.yml accordingly
3. pip install pyannote.db.callhome

instead of 

1. Download original files from talkbank.org
2. Run script to truncate and re-encode audio files
3. Edit ~/.pyannote/db.yml accordingly
4. pip install pyannote.db.callhome

What do you think?

**Comment by venkatesh-1729** [comment]
Hey @hbredin, so the reason why we did this was:
1. CallHome transcripts (who spoke when information) are not available for the entire file, hence we truncated them to the transcript length.
2. CallHome files are available in mp3 format and pyannote-audio doesn't play well with other codecs like mp3 etc. as wav format is assumed in many places like [here](https://github.com/pyannote/pyannote-audio/blob/master/pyannote/audio/features/with_librosa.py#L94-L95), [here](https://github.com/pyannote/pyannote-audio/blob/master/pyannote/audio/features/with_yaafe.py#L98-L99) and [here](https://github.com/pyannote/pyannote-audio/blob/master/pyannote/audio/applications/feature_extraction.py#L188), hence we encoded them to wav format.

In pyannote-audio we can add the feature to support multiple audio formats by using libraries like [pydub](https://github.com/jiaaro/pydub), then there is no need to re-encode files.

**Comment by hbredin** [comment]
1. This is the exact reason why the "annotated" entry exists. 
Even though, only a part of the file was manually annotated (and hence only this part can be evaluated), one may still want to process the whole file to achieve (hopefully better) performance on the annotated part. For instance, the [REPERE corpus](https://github.com/pyannote/pyannote-db-repere/blob/master/REPERE/data/speaker/all.trn.uem) is also partially annotated.

2. A lot has changed in the `develop` branch with respect to this (I expect to make a new release in the coming weeks): 
    * `wav` has been changed into `audio`
    * `pysndfile` has been replaced by `librosa.load`

**Comment by venkatesh-1729** [comment]
@hbredin 
1. I'll update the mdtm, uem files accordingly without truncation
2. For now, we can make re-encoding optional and change it accordingly for the new release.

Great work, by the way 👍

**Comment by hbredin** [comment]
If I were you, I wouldn't even convert original `.cha` files to `.mdtm`.

I would use the `.cha` file directly to generator the `{'annotation': Annotation(), 'annotated'= Timeline()}` instances. 

This would save you some time later if you plan to add other protocols for the other parts of CallHome.
This would also remove the dependency to `pyannote.parser`.

**Comment by venkatesh-1729** [comment]
@hbredin I would like to keep the script for generating `.mdtm`, `.uem` from `.cha` files separate for usability purposes. 

1. Clean installation of the database, one can directly use the data in the database or generate new ones by running the script and install.
2. Don't have to meddle directory of `.cha` or `.wav` files in `Protocol` and `Database` classes. If we were to directly use `.cha` files to create `Annotation` and `Timeline` instances this is not possible. 

I hope #3 issues resolves the truncating issue but keeping re-encoding step until next release.

**Comment by hbredin** [comment]
Thanks. I'll have a closer look at it (i.e. actually test it) in the coming weeks.

2. I meant distributing the `.cha` files with the `pyannote.db.callhome` package, the same way you are already doing it with `.mdtm` files. I did not mean finding them automagically like we already do for audio files (this is what you meant, right?)

**Comment by venkatesh-1729** [comment]
@hbredin I would rather prefer downloading transcript and audio files separately than shipping them with the code.

**Comment by hbredin** [comment]
Ok. That's understandable. 
That might even be forbidden by the license terms for "CallHome". Is it?

**Comment by hbredin** [comment]
That being said, sharing `mdtm` files derived from `cha` file might actually also be forbidden by the license terms...

**Comment by venkatesh-1729** [comment]
I couldn't find any license for CallHome on the official website though.

**Comment by hbredin** [comment]
Had some time to think about this issue today...

What if the package itself automatically downloads the `.cha` files from Callhome webpage the first time is used? Since it is public anyway, this would make the package even easier to use?

What do you think?

**Comment by venkatesh-1729** [comment]
Yeah sounds good, then we don't need to worry about terms and conditions of the license. 
The pyannote package will come with either a script or hook which can download the required data from the public domain for creating `mdtm` files etc.
